client/components/domains/domain-search-results/index.jsx

```diff
@@ -106,7 +105,7 @@ class DomainSearchResults extends React.Component {
 					'{{small}}If you purchased %(domain)s elsewhere, you can {{a}}map it{{/a}} for free.{{/small}}',
 					{ args: { domain }, components }
 				);
-			} else if ( ! this.props.domainsWithPlansOnly || this.props.isSiteOnPaidPlan ) {
+			} else if ( ! domainsWithPlansOnly ) {
```

Comment:
We check if a mapping is free above with `isDomainMappingFree()`. So at this point, if not DWPO user, it will have the default cost.